csp/decorators.py

```diff
@@ -36,12 +66,33 @@ def _wrapped(*a, **kw):
     return decorator
 
 
-def csp(**kwargs):
-    config = dict(
-        (k.lower().replace('_', '-'), [v] if isinstance(v, str) else v)
-        for k, v
-        in kwargs.items()
-    )
+def csp(policies=(), policy_definitions=None, **kwargs):
+    policies = iter_policies(tuple(policies))
+
+    if kwargs and not isinstance(next(iter(kwargs.values())), dict):
+        # Single-policy kwargs is the legacy behaviour (deprecate?)
+        # TODO: single-policy format isn't currently supported for policy_dict
```

Comment:
TODO: single-policy format isn't currently supported for policy_dict

Should just be a matter of calling normalize?